src/mount/sauna_client.cc

```diff
@@ -878,9 +899,27 @@ AttrReply getattr(Context &ctx, Inode ino) {
 #ifdef _WIN32
 	patch_uid_gid_fields(o_stbuf);
 #endif
+
+	// If getattr succeeded and data did not come from cache, then cache it
+	if (!fromcache) {
+		auto data_acquire_time = gDirEntryCache.updateTime();
+
+		std::unique_lock<shared_mutex> write_guard(gDirEntryCache.rwlock());
+		gDirEntryCache.updateTime();
+
+		gDirEntryCache.insert(ctx, ino, attr, data_acquire_time);
+		if (gDirEntryCache.size() > gDirEntryCacheMaxSize) {
+			gDirEntryCache.removeOldest(gDirEntryCache.size() -
+			                            gDirEntryCacheMaxSize);
+		}
+
+		write_guard.unlock();
```

Comment:
This lock should be automatically released here. Please consider removing the explicit unlock.